File: CS50/pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int l = 0, w = 0, s = 0;
    char head, nextHead;
    bool firstWord = true;
    string text = get_string("Text: ");
    int n = strlen(text);
    for (int i = 0 ; i < n ; i++)
    {
        head = text[i];
        nextHead = text[i + 1];
        if (isupper(head) || islower(head))
        {
            l++;
            if (nextHead == '.' || nextHead == '!' || nextHead == '?')
            {
                s++;
            }
            if (firstWord)
            {
                w++;
                firstWord = false;
            }
        }
        else if (isspace(head) && (isalpha(nextHead) || nextHead == '\"'))
        {
            w++;
        }
    }
    if (s == 0)
    {
        s++;
    }
    printf("L: %i\nW: %i\nS: %i\n", l, w, s);
    float L = (l / (float)w) * 100;
    float S = (s / (float)w) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: CS50/pset6/readability.py
l, w, s = 0, 0, 0
head, nextHead = str(), str()
firstWord = True

text = input('Text: ')

for i in range(len(text) - 1):
    head = text[i]
    nextHead = text[i+1]
    if head.isalpha():
        l += 1
        if nextHead in ['.', '!', '?']:
            s += 1
        if firstWord:
            w += 1
            firstWord = False
    elif head.isspace() and (nextHead.isalpha() or nextHead == '\"'):
         w += 1

if s == 0:
    s += 1

print(f'L: {l}\nW: {w}\nS: {s}\n')

L = l / w * 100
S = s / w * 100
index = round(0.0588 * L - 0.296 * S - 15.8)
if index > 16:
    print('Grade 16+')
elif index < 1:
    print('Before Grade 1')
else:
    print(f'Grade {index}')